nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/upload/RUploadHandler.java

```diff
@@ -111,4 +114,11 @@ public VariableResolverAdapter getVariableResolverAdapter() {
   public ContentPermissionChecker contentPermissionChecker() {
     return contentPermissionChecker;
   }
+
+  private void validateUploadRequest(final String path) {
+    if (!isValidArchivePath(path)) {
+      throw new ValidationErrorsException("Non-R archive extension or wrong upload path.");
```

Comment:
Suggestion: Change from "Non-R archive extension" to "Extension not zip, tar.gz or tgz, or wrong upload path", as that is more helpful to someone looking at it without deep knowledge of R.

Reply:
Nice suggestion, done in d659b9bb95d8f2914780f13608efcb4d1f2d83b9